src/scf-release/jobs/wait-for-api/templates/pre-start.erb

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash -e
+
+# Report progress to the user; use as printf
+status() {
+    local fmt="${1}"
+    shift
+    printf "\n%b${fmt}%b\n" "\033[0;32m" "$@" "\033[0m"
+}
+
+# Report problem to the user; use as printf
+trouble() {
+    local fmt="${1}"
+    shift
+    printf "\n%b${fmt}%b\n" "\033[0;31m" "$@" "\033[0m"
+}
+
+# helper function to retry a command several times, with a delay between trials
+# usage: retry <max-tries> <delay> <command>...
+function retry () {
+    max=${1}
+    delay=${2}
+    i=0
+    shift 2
+
+    while test ${i} -lt ${max} ; do
+        printf "Trying: %s\n" "$*"
+        if "$@" ; then
+            status ' SUCCESS'
+            break
+        fi
+        trouble '  FAILED'
+        status "Waiting ${delay} ..."
+        sleep "${delay}"
+        i="$(expr ${i} + 1)"
+    done
+}
+
+API_ENDPOINT="http://<%= p("cc.internal_service_hostname") %>:<%= p("cc.external_port") %>/v2/info"
```

Comment:
We can still use `API_ENDPOINT="<%= p("cc.external_protocol") %>://<%= p("cc.internal_service_hostname") %>:<%= p("cc.external_port") %>/v2/info"`. But `cc.external_port` should be 9024. I think it is a bug that the default `cc.external_protocol` is https and the `cc.external_port` is 9022 (which is plain HTTP).

Reply:
Ref commit 45549937df903670bb4a651560b06fa23532afe2 (contrary to message, port 9024 is used). Pushed for jenkins to have a go after a rebase.

Reply:
`git rebase -i HEAD~` then choose `reword`. It should do the trick to change from 9023 to 9024 in the commit message.

Reply:
Yes, in the plan.